extensions/stackdriver/stackdriver.cc

```diff
@@ -472,6 +497,48 @@ StackdriverRootContext* StackdriverContext::getRootContext() {
   return dynamic_cast<StackdriverRootContext*>(root);
 }
 
+void StackdriverRootContext::logTCPOpenOnTimeout(
+    StackdriverRootContext::TcpRecordInfo& record_info,
+    const std::string& destination_namespace, bool outbound,
+    const ::Wasm::Common::FlatNode& peer_node) {
+  auto cur = static_cast<long int>(
+      proxy_wasm::null_plugin::getCurrentTimeNanoseconds());
+  ::Wasm::Common::RequestInfo& request_info = *record_info.request_info;
+  // Attempt to log TCP Open Event if tcp open request log has timed out and it
+  // has not been logged before.
+  if (enableServerAccessLog() && !record_info.tcp_open_log_entry_logged &&
+      ((cur - request_info.start_time) > tcp_log_entry_timeout_)) {
+    // Populate RequestInfo with local node information.
+    ::Wasm::Common::populateTCPRequestInfo(
+        outbound, record_info.request_info.get(), destination_namespace);
+    ::Wasm::Common::populateExtendedRequestInfo(record_info.request_info.get());
+    logTCPOpen(record_info, peer_node);
+    // Set is_populated to false, so that populateTCPRequestInfo can be called
+    // again when peer metadata is available by RecordTCP.
+    record_info.request_info->is_populated = false;
+  }
+}
+
+void StackdriverRootContext::logTCPOpen(
+    StackdriverRootContext::TcpRecordInfo& record_info,
+    const ::Wasm::Common::FlatNode& peer_node) {
+  // Log TCP Open event log, if it has not been logged before.
+  if (!record_info.tcp_open_log_entry_logged) {
```

Comment:
Nit: early return here and move the logic out of the block:
```
if (record_info.tcp_open_log_entry_logged) {
  return;
}
```

Reply:
Removed now